Sources/SwiftDocC/SwiftDocCMarkdownOutput/MarkdownOutputManifest.swift

```diff
@@ -0,0 +1,98 @@
+@_spi(MarkdownOutput)
+public struct MarkdownOutputManifest: Codable, Sendable {
+    public static let version = "0.1.0"
+    
+    /// The version of this manifest
+    public let manifestVersion: String
+    /// The manifest title, this will typically match the module that the manifest is generated for
+    public let title: String
+    /// All documents contained in the manifest
+    public var documents: Set<Document>
+    /// Relationships involving documents in the manifest
+    public var relationships: Set<Relationship>
+    
+    public init(title: String, documents: Set<Document> = [], relationships: Set<Relationship> = []) {
+        self.manifestVersion = Self.version
+        self.title = title
+        self.documents = documents
+        self.relationships = relationships
+    }
+}
+
+extension MarkdownOutputManifest {
+    
+    public enum DocumentType: String, Codable, Sendable {
+        case article, tutorial, symbol
+    }
+    
+    public enum RelationshipType: String, Codable, Sendable {
+        /// For this relationship, the source URI will be the URI of a document, and the target URI will be the topic to which it belongs
+        case belongsToTopic
+        /// For this relationship, the source and target URIs will be indicated by the directionality of the subtype, e.g. source "conformsTo" target. 
+        case relatedSymbol
+    }
+    
+    /// A relationship between two documents in the manifest.
+    ///
+    /// Parent / child symbol relationships are not included here, because those relationships are implicit in the URI structure of the documents. See ``children(of:)``.
+    public struct Relationship: Codable, Hashable, Sendable {
+        
+        public let sourceURI: String
+        public let relationshipType: RelationshipType
+        public let subtype: String?
+        public let targetURI: String
+        
+        public init(sourceURI: String, relationshipType: MarkdownOutputManifest.RelationshipType, subtype: String? = nil, targetURI: String) {
+            self.sourceURI = sourceURI
+            self.relationshipType = relationshipType
+            self.subtype = subtype
+            self.targetURI = targetURI
+        }
+    }
+    
+    public struct Document: Codable, Hashable, Sendable {
+        /// The URI of the document
+        public let uri: String
+        /// The type of the document
+        public let documentType: DocumentType
+        /// The title of the document
+        public let title: String
+                
+        public init(uri: String, documentType: MarkdownOutputManifest.DocumentType, title: String) {
+            self.uri = uri
+            self.documentType = documentType
+            self.title = title
+        }
+        
+        public func hash(into hasher: inout Hasher) {
+            hasher.combine(uri)
+        }
+    }
+    
+    public func children(of parent: Document) -> Set<Document> {
```

Comment:
Question: What is this meant to represent?

Reply:
Please note that the added comment doesn't fully answer my question. 

- What does it mean for a "document" to be the "parent" of another document? 
  - Is this based on the developer defined documentation hierarchy or is it based on the base symbol hierarchy?
- What is a caller meant to use this function for?
  - If the caller is meant to process the document's hierarchically, would it make more sense to have the data also have a hierarchy?